components/Common/Breadcrumbs/index.tsx

```diff
@@ -0,0 +1,75 @@
+import classNames from 'classnames';
+import { type LinkProps } from 'next/link';
+import { useMemo, type FC } from 'react';
+
+import LocalizedLink from '@/components/LocalizedLink';
+
+import BreadcrumbItem from './BreadcrumbItem';
+import BreadcrumbRoot from './BreadcrumbRoot';
+import styles from './index.module.css';
+
+type BreadcrumbLink = {
+  label: string;
+  href: LinkProps['href'];
+};
+
+type BreadcrumbsProps = {
+  links: BreadcrumbLink[];
+  maxLength?: number;
+  hideHome?: boolean;
+};
+
+const Breadcrumbs: FC<BreadcrumbsProps> = ({
+  links = [],
+  maxLength = 3,
+  hideHome = false,
+}) => {
+  const items = useMemo(() => {
+    const itemsToRender = links.slice(
+      links.length > maxLength ? -maxLength : 0
+    );
+
+    return itemsToRender.map((link, index, items) => {
+      const isLastItem = index === items.length - 1;
+      let position = index + 1;
+      if (!hideHome) {
```

Comment:
Hi @ovflowd, I've gathered some insights from other design systems / Breadcrumb implementations.
To be honest, I am not much of an expert in accessibility / SEO in general, so I am not too sure what is the best way to implement an accessible + SEO friendly breadcrumbs, but here might provide you with some ideas:

## Atlassian

[Demo](https://atlassian.design/components/breadcrumbs/examples#long-breadcrumbs)

- Javascript rendering


## Carbon Design System

[Demo](https://react.carbondesignsystem.com/?path=/story/components-breadcrumb--breadcrumb-with-overflow-menu)

- Javascript rendering
- Render popover menu within the breadcrumb item
- Popover `ul` with `aria-label="Overflow menu in a breadcrumb"`

## React Spectrum (Adobe)

[Demo](https://react-spectrum.adobe.com/react-spectrum/Breadcrumbs.html#visible-items-overflow-behavior)

- Javascript rendering
- Additional Breadcrumbs rendered within a React Portal
- `…` breadcrumb item with `aria-label="…"`
- Menu popover with `aria-labelledby` the id of the button that triggers the popup.

## Dell

[Demo](https://vanilla.delldesignsystem.com/2.34.0/iframe.html?viewMode=docs&id=components-breadcrumb--basic#seo-guidelines)

- Hide truncated breadcrumbs with CSS
- `…` breadcrumb item with `aria-label="show 2 more navigation levels"`
- Expands when clicked on `…` breadcrumb item without popover menu

---

I believe there might be more examples out there but this should give some good insights.

Overall, everyone seem to implement it differently. `Dell` even provided some best practices for SEO / Accessibility concern that might worth a look.

Let me know what you think!

Reply:
Right, I'm not sure if what I mentioned made sense or was well explained.

What I'm trying to say is that we should remove the array slice, and hide the element via CSS.

Otherwise the Breadcrumbs will always be different which will confuse SEO. Because with JavaScript rendering and omitting certain entries they will simply not be in the DOM at all.

And we are only hiding those extra entries for "fitting in the screen purpose". But by doing this way, we're right now having this hacky "position += 1" and other parts of the logic that are honestly not cool at all.

So you can remove all this extra "garbage" code, and remove the array slice and everything. And simply render everything. And for the items you don't want to render (because currentIndex > maxLength) then you do:

```tsx
className({ "tailwind classname for hiding element": index > maxLength })
```

Reply:
Yes, I understood what you meant in your first comment. I was just wondering if we want to explore further before making a decision as currently there are different implementations to the same issue.

Reply:
Afaik, the code changes would not change how the component works (as initially described by the issue) and what is rendered and whatnot. I would rather have this simple change be done, and you can create a follow up issue with your concerns and different implementation ideas and we can go through it.

I even wonder if we could somehow "grab/reuse" the logic for conditional `...` that is being introduced on the Pagination component.

We can always iterate this again after getting the first version merged, and then we can verify if we should update the logic/simplify/improve etc